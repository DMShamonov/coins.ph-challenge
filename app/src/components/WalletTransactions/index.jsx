// @flow

import * as React from 'react';

import {
  Table,
  TableHeader,
  TableBody,
  TableRow,
  TableCell,
} from 'components/Table';
import WalletTransaction from 'components/WalletTransaction';

import type { PropsType } from './types.js.flow';

export default function WalletTransactions({
  walletId,
  walletTransactions,
}: PropsType): React.Node {
  return (
    <Table>
      <TableHeader>
        <TableRow>
          <TableCell>Date</TableCell>
          <TableCell width="10%">Fee</TableCell>
          <TableCell width="10%">Value</TableCell>
          <TableCell width="10%">Status</TableCell>
        </TableRow>
      </TableHeader>
      <TableBody>
        {walletTransactions.map(({
          id,
          date,
          fee,
          entries,
          pending,
        }) => (
          <WalletTransaction
            date={new Date(date).toISOString()}
            fee={fee}
            value={entries.find(({ account }) => account === walletId).value}
            pending={pending}
            key={id}
          />
        ))}
      </TableBody>
    </Table>
  );
}
