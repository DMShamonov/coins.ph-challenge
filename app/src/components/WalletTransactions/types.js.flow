// @flow

import type { WalletTransactionType } from 'types.js.flow';

export type PropsType = {|
  walletId: string,
  walletTransactions: WalletTransactionType,
|};
