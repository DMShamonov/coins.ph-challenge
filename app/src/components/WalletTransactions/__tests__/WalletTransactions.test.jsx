import React from 'react';
import { shallow } from 'enzyme';

import WalletTransactions from '../index';

describe('`WalletTransaction` component should render correctly', () => {
  test('with default props', () => {
    expect(shallow((
      <WalletTransactions
        walletId="__WALLET_ID__"
        walletTransactions={[{
          id: '__ID__',
          date: '2019-04-09T02:56:04.152Z',
          entries: [{ account: '__WALLET_ID__', value: 100000000 }],
          fee: 1000,
          pending: false,
        }]}
      />
    ))).toMatchSnapshot();
  });

  test('when `walletTransactions` prop is empty array', () => {
    expect(shallow(<WalletTransactions walletId="__WALLET_ID__" walletTransactions={[]} />)).toMatchSnapshot();
  });
});
