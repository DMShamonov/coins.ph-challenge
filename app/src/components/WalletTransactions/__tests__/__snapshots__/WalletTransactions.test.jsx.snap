// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`\`WalletTransaction\` component should render correctly when \`walletTransactions\` prop is empty array 1`] = `
<Table>
  <TableHeader>
    <TableRow>
      <TableCell>
        Date
      </TableCell>
      <TableCell
        width="10%"
      >
        Fee
      </TableCell>
      <TableCell
        width="10%"
      >
        Value
      </TableCell>
      <TableCell
        width="10%"
      >
        Status
      </TableCell>
    </TableRow>
  </TableHeader>
  <TableBody />
</Table>
`;

exports[`\`WalletTransaction\` component should render correctly with default props 1`] = `
<Table>
  <TableHeader>
    <TableRow>
      <TableCell>
        Date
      </TableCell>
      <TableCell
        width="10%"
      >
        Fee
      </TableCell>
      <TableCell
        width="10%"
      >
        Value
      </TableCell>
      <TableCell
        width="10%"
      >
        Status
      </TableCell>
    </TableRow>
  </TableHeader>
  <TableBody>
    <WalletTransaction
      date="2019-04-09T02:56:04.152Z"
      fee={1000}
      key="__ID__"
      pending={false}
      value={100000000}
    />
  </TableBody>
</Table>
`;
